### ISSUE TITLE
TechDraw: Print All Pages gives you a bitmap image (Windows specific)

### ISSUE THREAD

**Issue by Pendelf** [FreeCAD/FreeCAD#18235]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

You get a bitmap image when you use the “Print All Pages” function

https://github.com/user-attachments/assets/88603405-a41f-4c3c-9f02-585caa71f9df

![Снимок экрана 2024-12-02 012644](https://github.com/user-attachments/assets/2fbfe5b1-8d1d-4462-a26d-9a676c305432)

### Full version info

```shell
OS: Windows 11 build 26100
Architecture: x86_64
Version: 1.1.0dev.39380 (Git) Conda
Build type: Release
Branch: main
Hash: 797c1180b012a8aee1d67de62880df0f835a731e
Python 3.11.10, Qt 5.15.15, Coin 4.0.3, Vtk 9.3.0, OCC 7.8.1
Locale: Russian/Russia (ru_RU)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/Fusion
Installed mods: 
  * Curves 0.6.51
  * fasteners 0.5.31
  * PieMenu 1.9.5
  * sheetmetal 0.5.6
```

### Subproject(s) affected?

TechDraw

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Pendelf** [comment]
Screenshot from Inkscape

![Снимок экрана 2024-12-02 013752](https://github.com/user-attachments/assets/535e9764-f87e-4f12-a5e3-29cad9a78331)

**Comment by maxwxyz** [comment]
@WandererFan is this the intended behavior or a bug?

**Comment by WandererFan** [comment]
Didn't even know there were variations of pdf files.   

From a quick glance, the only difference I see between printPdf and printAllPdf is that printAllPdf sets the resolution of the QPdfWriter and printPdf uses the defaults (pdfWriter.setResolution(printer->resolution());).   No idea if that is relevant.

How do I tell if the pdf is bitmap or vector?

**Comment by Pendelf** [comment]
![Снимок экрана 2024-12-02 212149](https://github.com/user-attachments/assets/d777bd5b-6a15-4af9-b28a-ad8010aa6deb)

The files even have different weights. You can check if the file is vector or raster in a vector editor, for example, Inkscape or open pdf in LibreOffice

**Comment by mpfusion** [comment]
Just to let you know: I cannot reproduce the issue here. “Export PDF” and “Print All Pages” result in the same PDF both in vector format. And both with same file size.
```
OS: Debian GNU/Linux 12 (bookworm) (X-Cinnamon/lightdm-xsession/xcb)
Architecture: x86_64
Version: 1.1.0dev.39419 (Git)
Build type: Release
Branch: main
Hash: 7fdc3b807ac3f84ace7f1e42b7c2bc73939ed98c
Python 3.11.2, Qt 5.15.8, Coin 4.0.0, Vtk 9.1.0, OCC 7.6.3
Locale: English/Ireland (en_IE)
Stylesheet/Theme/QtStyle: unset/FreeCAD Classic/Qt default
Installed mods: 
  * fasteners 0.5.33
  * kicadStepUpMod 11.4.3
```
[export.pdf](https://github.com/user-attachments/files/17993747/export.pdf)
[print_all.pdf](https://github.com/user-attachments/files/17993749/print_all.pdf)

**Comment by Pendelf** [comment]
Tried it on Manjaro, everything works great

OS: Manjaro Linux (KDE/plasmawayland/xcb)
Architecture: x86_64
Version: 1.1.0dev.39416 (Git) Conda AppImage
Build type: Release
Branch: main
Hash: f6268ffd2877dadb4440ade541e12c388a517d6e
Python 3.11.9, Qt 5.15.13, Coin 4.0.3, Vtk 9.2.6, OCC 7.7.2
Locale: English/United States (en_US)
Stylesheet/Theme/QtStyle: unset/FreeCAD Dark/Qt default

**Comment by WandererFan** [comment]
Just another data point - removing the "setResolution"  from printAllPdf doesn't change the output on linux, still no bit maps

**Comment by WandererFan** [comment]
Does anybody know the details of PDF/A-1b? Could printAll and printAllPdf be producing different flavours of Pdf (PDF 1.4 vs PDF/A-1b)?   We just had an issue with transparency due to different flavours of Pdf.

**Comment by Pendelf** [comment]
Still relevant

**Comment by tobiasfalk** [comment]
Ping

**Comment by WandererFan** [comment]
How are you getting a pdf from PrintAllPages?  Via the OS's print-to-pdf option?  

When I compare a file from Export Pdf with a File > Print (OS print-to-pdf) I don't see any differences.  File > Export Pdf output looks the same too.

But. I'm on linux and the OP is on Win.

**Comment by Pendelf** [comment]
@WandererFan 
Yes, I tried it on Manjaro and everything works fine there, apparently this is somehow related specifically to Windows
https://github.com/FreeCAD/FreeCAD/issues/18235#issuecomment-2514864628

**Comment by WandererFan** [comment]
Note that Mac seems to have Pdf export issues as well: https://forum.freecad.org/viewtopic.php?t=103997

**Comment by WandererFan** [comment]
@wo9f , @Pendelf - finally managed to build a windows version with Qt6.11.  Could you check these 2 files please?

One uses MS Print to Pdf print engine, and the other uses QPdfWriter.   It seems to me that the MS Print to Pdf version still has bitmap tiles, while the QPdfWriter has vectors for the graphics, but still uses bitmap for the template (although I don't see tiles).  

PrintAll will use the MS Print to Pdf path in the code, so we probably want to change that to use QPdfWriter as a first step.

[print1_qt6.11_MSPrintPdf.pdf](https://github.com/user-attachments/files/27318698/print1_qt6.11_MSPrintPdf.pdf)
[exportPdf_qt6.11_QPdfWriter.pdf](https://github.com/user-attachments/files/27318697/exportPdf_qt6.11_QPdfWriter.pdf)

**Comment by tobiasfalk** [comment]
@WandererFan 
Interesting that the templates are not rendering correctly, could be connected how SVGs are imported, as I remember correctly the pathe of SVG import and template import works are very similar.
And if this is based on the QT bug in 6.8 it should be fixed by now.
[https://qt-project.atlassian.net/browse/QTBUG-123817](https://qt-project.atlassian.net/browse/QTBUG-123817)

**Comment by tobiasfalk** [comment]
Could you maybe try to also import a SVG, with text and then export/print the PDFs.

If that is indeed the case it is related to: https://github.com/FreeCAD/FreeCAD/issues/21096
Which is already mentioned on both sides, quite often.

**Comment by wo9f** [comment]
@WandererFan 
Hi,
I’ve checked both PDF files using Adobe Acrobat Pro and used the layer analysis feature in the prepress tools.
The results are attached.
The file generated with QPdfWriter appears to be fully vectorized (see screenshot).
In contrast, the file generated via MS Print to PDF is rasterized (also see screenshots).
So from my side, QPdfWriter produces the expected vector output, while MS Print to PDF converts the content into bitmaps.

One additional question:
it would be interesting to check how QPdfWriter handles rotated text in callouts (for example at 45° or 60°).
Will such text remain as vector/text objects in the PDF, or will it be rasterized?
Could you please verify this case as well?

[print1_qt6.11_MSPrintPdf_layer.pdf](https://github.com/user-attachments/files/27332736/print1_qt6.11_MSPrintPdf_layer.pdf)
[exportPdf_qt6.11_QPdfWriter_layer.pdf](https://github.com/user-attachments/files/27332737/exportPdf_qt6.11_QPdfWriter_layer.pdf)

<img width="1728" height="857" alt="Image" src="https://github.com/user-attachments/assets/e69310d4-06cd-44a9-881e-506cdc4ce9cd" />
<img width="883" height="640" alt="Image" src="https://github.com/user-attachments/assets/a668b854-0e28-4288-b802-60bd53fb2bc5" />
<img width="1732" height="855" alt="Image" src="https://github.com/user-attachments/assets/7a6b81c0-2be6-4a30-b67d-71b3e2ab3499" />
